File: ruby/get_block.rb
# ruby demo_get_block.rb

require 'solana_rpc_ruby'
require 'pry'
require 'dotenv'
require 'time'

Dotenv.load
rpc_url = ENV['RPC_URL']

SolanaRpcRuby.config do |c|
  c.json_rpc_version = '2.0'
  c.cluster = rpc_url
end

method_wrapper = SolanaRpcRuby::MethodsWrapper.new(cluster: SolanaRpcRuby.cluster)

puts "slot      TX-count"
110771234.upto(110771235).each do |slot|
  begin
    block = method_wrapper.get_block(slot)
  rescue SolanaRpcRuby::ApiError
    block = nil
  end

  tx_count = block.nil? ? 0 : block.result['transactions'].count

  # byebug
  puts "#{slot} #{tx_count}"
  sleep(1)
end

File: js/get_block.mjs
// This script will fetch the most recent finalized block and 
// return the slot number, elapsed time, and transaction count.
//
// Provide RPC_URL in your .env file to check your connection.
//
// Example Use:
// node get_block.mjs

import * as web3 from '@solana/web3.js';
import dotenv from 'dotenv';
dotenv.config();

// Set up web3 client
const connection = new web3.Connection(
  process.env.RPC_URL, 'finalized'
);

// Get the latest finalized slot
const slot = await connection.getSlot('finalized');

let startTimestamp = new Date().getTime();
// Fetch the block for that slot
const block = await connection.getBlock(
  slot, 
  { 
    commitment: 'finalized', 
    maxSupportedTransactionVersion: 1 
  }
);
let endTimestamp = new Date().getTime();

console.log(`${new Date().toISOString()}: Slot: ${slot}, tx: ${block.transactions.length}, ms: ${endTimestamp - startTimestamp}`);